File: dev-mode/run-support/src/main/scala/play/runsupport/NamedURLClassLoader.scala
/*
 * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package play.runsupport

import java.net.{ URL, URLClassLoader }

/**
 * A ClassLoader with a toString() that prints name/urls.
 */
class NamedURLClassLoader(name: String, urls: Array[URL], parent: ClassLoader) extends URLClassLoader(urls, parent) {
  override def toString = name + "{" + getURLs.map(_.toString).mkString(", ") + "}"
}

File: dev-mode/play-run-support/src/main/java/play/runsupport/classloader/NamedURLClassLoader.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.runsupport.classloader;

import java.net.URL;
import java.net.URLClassLoader;
import java.util.Arrays;

/** A ClassLoader with a toString() that prints name/urls. */
public class NamedURLClassLoader extends URLClassLoader {

  private final String name;

  public NamedURLClassLoader(String name, URL[] urls, ClassLoader parent) {
    super(urls, parent);
    this.name = name;
  }

  @Override
  public String toString() {
    return name + Arrays.toString(getURLs());
  }
}
